### PR TITLE
make the PMD rules better, more explicit and more opinionated

### PR DESCRIPTION
**Pull request by grimreaper**
Lets argue about all the rules :)

### CODE REVIEW

**Comment by thpr** [comment]
Having now tested this, I'm okay with it, but will let Connor check as well

File: code/standards/pmdruleset.xml [hunk 1]
```diff
@@ -15,23 +15,60 @@
...
+  <rule ref="rulesets/java/basic.xml">
+    <exclude name="AvoidUsingHardCodedIP" />
+  </rule>
+  <rule ref="rulesets/java/braces.xml" />
+  <rule ref="rulesets/java/clone.xml">
+    <exclude name="ProperCloneImplementation" />
+    <exclude name="CloneThrowsCloneNotSupportedException" />
+  </rule>
+  <rule ref="rulesets/java/codesize.xml" >
+    <exclude name="TooManyFields" />
```
Comment (thpr) [inline code/standards/pmdruleset.xml]: I actually disagree that we should suppress TooManyFields and TooManyMethods

Comment (grimreaper) [inline code/standards/pmdruleset.xml]: Does this provide value? I've found that the number of fields is rarely an issue, but the complexity and breakdown is. I'll remove the exclusions though (this PR is still strictly better than where we are today)

File: code/standards/pmdruleset.xml [hunk 2]
```diff
@@ -15,23 +15,60 @@
...
+    <exclude name="ConsecutiveLiteralAppends" />
+    <exclude name="AvoidStringBufferField" />
+  </rule>
+  <!--<rule ref="rulesets/java/sunsecure.xml">-->
+  <rule ref="rulesets/java/typeresolution.xml" />
+  <rule ref="rulesets/java/unnecessary.xml">
+    <exclude name="UselessParentheses" />
+  </rule>
+  <rule ref="rulesets/java/unusedcode.xml">
+    <exclude name="UnusedModifier" />
```
Comment (grimreaper) [inline code/standards/pmdruleset.xml]: that is included by default. Note these are *exclude* rules.

File: code/standards/pmdruleset.xml (hunk 1 above)
Comment (grimreaper) [inline code/standards/pmdruleset.xml]: exclusion removed

**Pull request merged**